### PR TITLE
Add release and branching doc

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: release_and_branching.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+We release set of operators as a single product called the 'OpenStack Operator'. The openstack-operator itself 
+acts as a meta operator. It contains CRDs which can be used to create an OpenStackControlplane in a single action.
+Additionally, installation of of a specific version of the openstack-operator drives the installation of all
+openstack service operators via OLM (Operator Lifecycle Manager) dependencies. This document aims to summarize the
+relationship between how these components are released and branched upstream.
+
+## Artifacts
+
+For each operator/repository upstream we build 3 artifict/containers:
```
Comment (abays) [inline release_and_branching.md:13]: `s/artifict/artifact`

File: release_and_branching.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+* A csv-merger is used to merge in the CSV from the dataplane-operator along with all the ENV variables harvested from the above.
+
+NOTE: The dataplane-operator is merged into the CSV of the openstack-operator so that it's CRD can be treated
+as a top level visible resource in the OpenShift UI.
+
+NOTE: We no longer merge all the CSV for all openstack service operators because we hit the bundle size limit. This is a limitation due to how OLM relies on ConfigMaps when unpacking bundles. (separate bundles are required for OpenStack operators at this time)
+
+## How go.mod is used to pin the go dependencies and control the service operator containers that get installed
+
+Each service operator that is integrated into the openstack-operator has an entry in the go.mod for the openstack-operator.  We required the use of ([psuedoversions] https://go.dev/ref/mod#pseudo-versions) in the go.mod file for all openstack service operator dependencies. In addition to controlling how the golang dependency gets installed the psuedoversion controls which bundle is used and installed with the openstack-operator and acts as a means to control and promote new service operator builds to be used.
```
Comment (abays) [inline release_and_branching.md:50]: The link here should use this format: `[display text](url)`

File: release_and_branching.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+In this case: quay.io/openstack-k8s-operators/cinder-operator-bundle:3c5c40e6cc3aba6e287b94d26770519f76ae9528 would get used.
+
+## Tagging submodules for api/apis (nested modules)
+
+Each service operator has either an 'api' or an 'apis' directory that contains the API structs, webhooks, and some common helpers for that operator. These directories have a separate nested go.mod file in order to minimimize the dependencies involved in using
+the CRD structs in external operators.
+
+NOTE: there are tentative plans to combine things into a common 'api' project at a later date.
+
+The api submodule directories can be periodically tagged (independant from the top level go.mod for an operator) to signify
```
Comment (abays) [inline release_and_branching.md:69]: `s/independant/independent`

File: release_and_branching.md (hunk 1 above)
Comment (dprince) [reply release_and_branching.md:13]: got it fixed

File: release_and_branching.md (hunk 2 above)
Comment (dprince) [reply release_and_branching.md:50]: you know, I tried creating this with vscode and see what happens. fixed, and vim it is next time

File: release_and_branching.md (hunk 3 above)
Comment (dprince) [reply release_and_branching.md]: fixed

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**